src/implementation/Server/HTTPServer/HTTPServerImpl.ts

```diff
@@ -27,9 +27,9 @@ export default class HTTPServerImpl {
         this.pubSubSubscriptionRoutes = [];
     }
 
-    registerPubSubSubscriptionRoute(pubSubName: string, topicName: string, route: string, metadata: KeyValueType): void {
+    registerPubSubSubscriptionRoute(pubSubName: string, topicName: string, route: string, metadata?: KeyValueType): void {
         const httpMetadata: { [key: string]: string } = {};
-        for (const [key, value] of Object.entries(metadata)) {
+        for (const [key, value] of Object.entries(metadata ?? {})) {
```

Comment:
Not needed as it should be set correctly in the method signature and validated straight under it